docs/src/adr/004adr-annotations.md

```diff
@@ -0,0 +1,136 @@
+# ADR-004: Syntax for Java-like annotations in comments
+
+**author:** Igor Konnov
+
+This ADR documents our decision on using Java-like annotations in comments.
+Our main motivation to have annotations is to simplify type annotations, as
+presented in [ADR-002][]. Hence, in the following text, we are using
+examples for type annotations. However, the annotations framework is not
+restricted to types. Similar to Java and Scala, we can use annotations
+to decorate operators with hints, which may aid the model checker.
+
+## 1. What can be annotated
+
+Annotations should be written in comments that are written in front of a
+declaration. The following declarations are supported:
+ 
+ 1. Constant declarations, e.g., `CONSTANT N`.
+ 1. Variable declarations, e.g., `VARIABLE x`.
+ 1. Top-level operator declarations, e.g., `Foo(x) == e`.
+ 1. LET-IN operator declarations, e.g., `Foo(x) == LET Inner(y) == e IN f`.
+ 1. Recursive functions, e.g., `foo[i \in Int] == e`.
+ 1. Recursive operators, that are declared either at the top-level or by using
```

Comment:
This also seems like not it's own item, but something that is a sub-item under, respectively, top level operator declarations and LET-IN declarations.